docs/customizing.md

```diff
@@ -0,0 +1,238 @@
+2. Preferences for how Pulp is configured and deployed. In particular, which plugins to install.
+3. Variables to control which version of plugins are installed, for reproducibility.
+
+WRT to #2, the large data structure variables are `pulp_settings` & `pulp_install_plugins`.
+WRT to #3, the large data structure variable is `pulp_install_plugins`.
+
+Because the Pulp installer is composed of Ansible roles, you can use the variables for each of these roles to customize your Pulp installation.
+For example, if you want to specify firewall requirements, edit the corresponding firewall variables associated with the [pulp_webserver](https://pulp-installer.readthedocs.io/en/latest/roles/pulp_webserver/#pulp_webserver) role.
+Each role documents all the variables that it uses. Some variables are
+used by multiple roles. In that case, they are documented in their primary role and mentioned in
+the `shared_variables` section the other roles.
+
+**Required Variables:**
+Most variables have generally compatible default values but a few are required (must be specified by the user). See ``playbooks/example-use/group_vars/all`` for
+the minimal set of required variables.
+
+Example of setting a multitude of variables:
+--------------------------------------------
+In the following example, a multitude of variables are set:
+
+```
+  vars:
+    pulp_workers: 4
+    pulp_install_plugins:
+      pulp-container: {}
+      pulp-deb:
+        upgrade: true
+      pulp-file:
+        collectstatic: true
+        version: "9.8.7"
+    pulp_settings:
+      secret_key: secret
+      content_origin: "{{ pulp_webserver_disable_https | default(false) | ternary('http', 'https') }}://{{ ansible_fqdn }}"
+      azure_account_name: 'Storage account name'
+      azure_container: 'Container name (as created within the blob service of your storage account)'
+      azure_account_key: 'Key1 or Key2 from the access keys of your storage account'
+      azure_url_expiration_secs: 60
+      azure_overwrite_files: 'True'
+      azure_location: 'the folder within the container where your pulp objects will be stored'
+      default_file_storage: 'storages.backends.azure_storage.AzureStorage'
+```
+
+`pulp_workers`, is set to `4`, which means that 4 worker processes will be started for processing jobs on a multi-core system. It replaces the default value of 2.
+
+`pulp_install_plugins` does the following:
+* Directs the installer to install all 3 plugins, pulp-deb, pulp-file and pulp-container.
+* If pulp-container is not installed, the latest version will be installed.
+* If pulp-container is already installed, the current version will be left intact.
+* If pulp-deb is not installed, the latest version will be installed.
+* If pulp-deb is already installed, it will be upgraded to the latest version.
+* If pulp-file is not installed, version `9.8.7` will be installed.
+* If pulp-file is already installed, it will be upgraded to `9.8.7`.
```

Comment:
We don't prevent a downgrade here, do we?
We also don't perform it (as in roll back the neccessary migrations).
Just a thought, maybe an actual issue.

Reply:
Yeah, that's something we haven't addressed yet from either an application perspective or an installer perspective.

I didn't want to mention it at all until we agree on a technical approach, or a message about the limitation.